### ISSUE TITLE
Update Image Explorer with Mentioned Enhancements

### ISSUE THREAD

**Issue by ZaxLofful** [rescuezilla/rescuezilla#303]
There was mention about having image explorer work with other formats.

I used ZST to make copies (very small) of my latest hard drives and I thought I would be able to use image explorer to view them and get a quick file (since there is no way to do this on Windows either), but I am not allowed to....Even tho I made the backup with Rescuezilla.

Now I have to restore 512GB to a random hard drive, just to get a single file off the image.

Please, update the ISO to allow viewing of Images and support other file formats....This is mentioned within the ISO itself, on the Image Explorer page.

**Comment by shasheene** [comment]
Yep, the 'Image Explorer (beta)' mode hasn't been very useful so far.

**I will do a big overhaul of it for the next release, and focus on adding [_indexed_ zstandard and even more importantly _indexed_ gzip](https://github.com/rescuezilla/rescuezilla/issues/153) to make things fast and useful.**

By the way, by other file formats I meant things like virtual machine formats, the other partclone-based imaging tools etc rather than other compression formats.

Also I recently saw a review of Rescuezilla that read:

> Even if I make uncompress backup, if I later do explore image it is impossible to copy any file because file manager can't show other attached devices during exploring mount.

So I really also need to implement [task #160 (add page to Image Explorer to select and mount a destination partition](https://github.com/rescuezilla/rescuezilla/issues/260).

I am aware there are a few compatibility isues with partclone-nbd/partclone-utils, and I've been meaning to [switch](https://github.com/rescuezilla/rescuezilla/issues/185) to the more popular partclone-utils and start fixing it.

**Comment by ZaxLofful** [comment]
1. I definitely think that the indexed archives will be of great help. This alone would actually directly solve the problem I have in this ticket.
2. I like the idea of virtual machine formats as well, since many times that is what I am using this for.
3. It would be nice to support rebuild and recover of ZFS/BRTFS partitions. I know this is new and not super easy, but it should be possible.

**Comment by ZaxLofful** [comment]
Any progress?

**Comment by shasheene** [comment]
Improving Image Explorer is something I intend to complete during my upcoming time-off work

**Comment by vasi** [comment]
Hi! I just wanted to point out that I wrote a tool awhile ago that uses FUSE to mount compressed formats for random access: https://github.com/vasi/lzopfs . It currently supports xz, lzop, bzip2 and gzip, and attempts to be as-indexed-as-possible. It's BSD-licensed, so pelase feel free to steal any code that's helpful.

Let me know if this is useful to you. I'm not sure if it's any better/worse than ratarmount for your purposes.

**Comment by stgiga** [comment]
@shasheene I used Zstd and soon I may no longer have the same computer (it's dying in its warrantied SSD) so a restore would be bad.

**Comment by shasheene** [comment]
I wrote an indexed-gzip plugin here https://gitlab.com/nbdkit/nbdkit/-/merge_requests/105

It will form the basis for a usable "Image Explorer" feature for gzip, and integrated into Rescuezilla shortly (hoped to do it this weekend, probably next weekend).

Writing an equivalent plugin for zstd will be required to close this issue. With AI tools, adding an equivalent zstd plugin to NBDKit is actually much less time-consuming than before, so despite this isue being years old the barriers for this task are rapidly falling.

**Comment by stgiga** [comment]
> I wrote an indexed-gzip plugin here https://gitlab.com/nbdkit/nbdkit/-/merge_requests/105
> 
> It will form the basis for a usable "Image Explorer" feature for gzip, and integrated into Rescuezilla shortly (hoped to do it this weekend, probably next weekend).
> 
> Writing an equivalent plugin for zstd will be required to close this issue. With AI tools, adding an equivalent zstd plugin to NBDKit is actually much less time-consuming than before, so despite this isue being years old the barriers for this task are rapidly falling.

@shasheene let me know when everything is all good. I'm hoping I can extract quite a few files that are only on the Zstd image of a computer I may soon no longer have.

**Comment by zaxlofful** [comment]
@shasheene : Any progress? Do you need testers for this? I would be happy to test it on the original archive I wanted this for. (I did end up doing the full restore, but would love it for my more recent ones)

**Comment by shasheene** [comment]
> Any progress

Yes some progress. Last month I tried used Claude Fable 5 to created a indexed-zstd "filter" for `nbdkit`, by referencing my [previous (hand-coded) index-gzip plugin](https://gitlab.com/nbdkit/nbdkit/-/merge_requests/105) for that program, and additionally referencing an existing C language indexed zstd tool: https://github.com/martinellimarco/libzstd-seek

I recall I had the AI generate a series of test cases and it passed them, I recall.

So now I have this branch with the changes: https://github.com/rescuezilla/nbdkit/tree/add-indexed-zstd-filter

To include this in Rescuezilla, the following is required:
1. Really thoroughly the validate the code of this new `indexed-zstd` filter (with more test cases and manual code-analysis)
2. make Rescuezilla build `nbkdit` from source (it currently includes it from Ubuntu repositories), itself first needs a separate important technical debt task first https://github.com/rescuezilla/rescuezilla/issues/150
3. include it in Rescuezilla frontend

And additionally raising the change to the author of nbdkit: https://gitlab.com/nbdkit/nbdkit
